crates/merod/src/cli/config.rs

```diff
@@ -44,53 +64,291 @@ impl FromStr for KeyValuePair {
+                stdout.set_color(ColorSpec::new().set_fg(Some(color)).set_intense(true))?;
+                write!(&mut stdout, "{}{}", sign, change)?;
+            }
+        }
 
+        stdout.reset()?;
         Ok(())
     }
 
-    pub async fn validate_toml(self, doc: &toml_edit::DocumentMut) -> EyreResult<()> {
-        let tmp_dir = temp_dir();
-        let tmp_path = tmp_dir.join(CONFIG_FILE);
+    fn print_config(&self, doc: &toml_edit::DocumentMut) -> EyreResult<()> {
+        match self.print {
+            PrintFormat::Default | PrintFormat::Toml => {
+                println!("{}", doc.to_string());
+            }
+            PrintFormat::Json => {
+                let value: serde_json::Value = toml::from_str(&doc.to_string())?;
+                println!("{}", serde_json::to_string_pretty(&value)?);
+            }
+            PrintFormat::Human => {
+                self.print_human(doc)?;
+            }
+        }
+        Ok(())
+    }
+
+    fn print_human(&self, doc: &toml_edit::DocumentMut) -> EyreResult<()> {
+        let mut table = Table::new();
+
+        table.load_preset("││──├─┤─┼─└ ┴┬┌ ┐");
+        table.set_header(vec![
+            Cell::new("Key").fg(ComfyColor::Blue),
+            Cell::new("Value").fg(ComfyColor::Green),
+            Cell::new("Type").fg(ComfyColor::Yellow),
+        ]);
 
-        write(&tmp_path, doc.to_string()).await?;
+        fn add_to_table(table: &mut Table, prefix: &str, item: &Item, schema: &ConfigSchema) {
+            match item {
+                Item::None => (),
+                Item::Value(value) => {
+                    table.add_row(vec![
+                        Cell::new(prefix),
+                        Cell::new(value.to_string()),
+                        Cell::new(schema.type_info),
+                    ]);
+                }
+                Item::Table(table_data) => {
+                    for (key, value) in table_data.iter() {
+                        let full_path = if prefix.is_empty() {
+                            key.to_owned()
+                        } else {
+                            format!("{}.{}", prefix, key)
+                        };
 
-        let tmp_path_utf8 = Utf8PathBuf::try_from(tmp_dir)?;
+                        if let Some(child_schema) = schema.find(&full_path) {
+                            if value.is_table() {
+                                add_to_table(table, &full_path, value, child_schema);
+                            } else {
+                                table.add_row(vec![
+                                    Cell::new(full_path),
+                                    Cell::new(value.as_str().unwrap_or("")),
+                                    Cell::new(child_schema.type_info),
+                                ]);
+                            }
+                        }
+                    }
+                }
+                Item::ArrayOfTables(_) => {
+                    // Handle arrays if needed
+                }
```

Comment:
yes, they are.

you can use `this.is.an.[0].item` to represent accessing them

sorta like `jq`, `jsonpath`